src/addrman.cpp

```diff
@@ -479,11 +479,15 @@ int CAddrMan::Check_()
 }
 #endif
 
-void CAddrMan::GetAddr_(std::vector<CAddress>& vAddr)
+void CAddrMan::GetAddr_(std::vector<CAddress>& vAddr, unsigned int max_addresses, unsigned int max_pct)
 {
-    unsigned int nNodes = ADDRMAN_GETADDR_MAX_PCT * vRandom.size() / 100;
-    if (nNodes > ADDRMAN_GETADDR_MAX)
-        nNodes = ADDRMAN_GETADDR_MAX;
+    unsigned int nNodes = vRandom.size();
+    if (max_pct != 0) {
```

Comment:
Seems more logical to use `>= 100` here?

Reply:
I think both are fine. Calling with 0 to indicate no max seems fine to me too.